### PR TITLE
do not modify content of comments

### PR DESCRIPTION
**Pull request by pums974**

### CODE REVIEW

**Comment by cphyc** [comment]
This is really great, thanks so much!
Note that at the moment, this conflicts with the rule defined https://github.com/cphyc/fortran-syntax/blob/60080449d2d7ea7b25b1a933ddcf0fada11c2f1c/fortran_linter/main.py#L54 which adds a space after a `!`.
```fortran
x = 1 !bad
x = 1 ! good
```

Do you think you could amend you PR to support this?

**Comment by pums974** [comment]
I'm sorry I had to change part of the logic here.
Now the corrections are applied only to the matched part of the line.

This is necessary to garantee that comments are untouched.

But this broke the current rules for spaces around operators, "::",  and "=",

Hence the large diff

File: fortran_linter/main.py [hunk 1]
```diff
@@ -52,7 +42,7 @@ class FortranRules:
         # Fix "foo! comment" to "foo ! comment"
         (r"(\w)\!", r"\1 !", "At least one space before comment"),
         # Fix "!bar" to "! bar"
-        (r"\!(|\s\s+)(\w)", r"! \2", "Exactly one space after comment"),
+        (r"\!(|\s\s+)([\S])", r"! \2", "Exactly one space after comment"),
```
Comment (cphyc) [inline fortran_linter/main.py:45]:
```suggestion
        (r"\!(|\s\s+)(\S)", r"! \2", "Exactly one space after comment"),
```

File: tests/test.f90 [hunk 2]
```diff
@@ -34,3 +34,6 @@ end module my_module
 do i=1,10,n
 
 x=1+2*3/5
+
+! "var=1"  ! Shouldn't be changed
```
Comment (cphyc) [inline tests/test.f90:38]:
```suggestion
!  "var=0"  ! only first space should be removed
! "var=1"  ! Shouldn't be changed
```

File: tests/test_reference.f90 [hunk 3]
```diff
@@ -34,3 +34,6 @@ end module my_module
 do i = 1, 10, n
 
 x = 1 + 2 * 3 / 5
+
+! "var=1"  ! Shouldn't be changed
```
Comment (cphyc) [inline tests/test_reference.f90:38]:
```suggestion
! "var=0"  ! only first space should be removed
! "var=1"  ! Shouldn't be changed
```

**Comment by cphyc** [comment]
Thanks very much for the PR!

**Comment by pums974** [comment]
Seems all good.
I just wonder if the rule "Exactly one space after comment" really make sense.

I often have block of code that are commented out, and this removes the indentation.

"At least one space after comment" would make more sense to me.
What do you think ?

**Comment by cphyc** [comment]
I guess this is a matter of preference. I personally favour
```fortran
x = 1     ! Define abscissa
y = x**2  ! Define ordinate
```
over
```fortran
x = 1 !    Define abscissa
y = x**2 ! Define ordinate
```
The reason is that, in the former example the `!` is grouped with the comment, while in the latter, it is grouped with the code. I personally see the `!` as part of the comment.

I would however be OK to have an opt-in preference to replace the "Exactly one space after comment" rule by "At least one space after comment", which could be set via the CLI.

**Comment by pums974** [comment]
I agree with your example, but this requires me to comment twice a block of code I want to keep around while keeping the indentation.

Not a big inconvenience though

**Pull request merged**